jedi/evaluate/context/iterable.py

```diff
@@ -54,6 +67,83 @@ def name(self):
 
 
 @has_builtin_methods
+class CoroutineMixin(object):
+    array_type = None
+
+    def get_filters(self, search_global, until_position=None, origin_scope=None):
+        gen_obj = compiled.create(self.evaluator, CoroutineType)
+        yield SpecialMethodFilter(self, self.builtin_methods, gen_obj)
+        for filter in gen_obj.get_filters(search_global):
+            yield filter
+
+    def py__bool__(self):
+        return True
+
+    def py__class__(self):
+        gen_obj = compiled.create(self.evaluator, CoroutineType)
+        return gen_obj.py__class__()
+
+    @property
+    def name(self):
+        return compiled.CompiledContextName(self, 'coroutine')
+
+
+class Coroutine(CoroutineMixin, Context):
+    def __init__(self, evaluator, func_execution_context):
+        if not HAS_COROUTINE:
+            raise ImportError("Need python3.5 to support coroutines.")
```

Comment:
I don't think we need to check for this. At the moment jedi is using the same version for autocompletion in the parser and type inference. It should therefor never be an issue.

Reply:
I'm actually working to get cross-Python support working for a lot of things. However that is an entirely different issue that you shouldn't care about. Just use the mindset that you're writing this for Python 3.6 and I'll help you to get the tests to pass. Just get the tests working for 3.6 :)

Reply:
Should I remove all version checks / ImportErrors or how do we proceed?

Reply:
Yes just remove them. :)